includes/publisher.php

```diff
@@ -232,6 +232,16 @@ public function create_post( $post_id, $item_slug, $item_props, $parent ){
             $stat_key = $new_post_id == $post_id ? 'updated' : 'new';
             $this->stats[ 'posts' ][ $stat_key ][ $new_post_id ] = get_post_permalink( $new_post_id );
 
+			// Upload the featured image, if any was set
+            // source: fullbright/git-it-write, commit d95887a087c7f5f854eed0f82ec68f151c72da74
+			$featured_image = $front_matter[ 'featured_image' ];
+			if(!empty( $featured_image )){
+				GIW_Utils::log( 'Uploading featured image ' . $featured_image );
+				$this->upload_featured_image($featured_image, $new_post_id);
```

Comment:
Can you please use `publisher.php` -> `upload_image()` function instead of a custom function to upload image? Also please try if the changes work as expected.